[incubator/anchore-engine] Adds incubator chart for Anchore Engine

zhill opened the pull request:
Adds Anchore Engine chart to incubator.

To test the service you can hit the API (as described in the NOTES.txt) or use the anchore-cli to query the system. The cli is available [here](https://github.com/anchore/anchore-cli) or via pip: `pip install anchorecli`. The instructions in NOTES indicates how to configure the cli.

zhill committed:
Adds incubator chart for Anchore Engine

zhill commented:
/assign @sameersbn

viglesiasce commented:
/assign

zhill commented:
Working on the linter issues in the CircleCI result. Will update with commit to resolve those.

zhill committed:
Fixes linter issues on anchore-engine values.yaml

zhill committed:
Fixes truth values for values.yaml

zhill committed:
Fixes for truthy values in other yaml configs for anchore-engine chart

zhill committed:
Fix ingress spec for anchore-engine chart for easier config with helm cli

viglesiasce commented:
/ok-to-test

viglesiasce commented on incubator/anchore-engine/Chart.yaml:
```diff
@@ -0,0 +1,19 @@
+   email: zach@anchore.com
+engine: gotpl
+icon: https://anchore.com/wp-content/uploads/2016/08/anchore.png
+appVersion: 0.1.3
```
nit, but can you move this to just under the version above.

viglesiasce commented on incubator/anchore-engine/README.md:
```diff
@@ -0,0 +1,83 @@
+=========================
+
+This chart deploys the Anchore Engine docker container image analysis system. Anchore Engine
+requires a PostgresSQL database (>=9.6) which may be handled by the chart of supplied externally,
```
/of/or/

viglesiasce commented on incubator/anchore-engine/templates/NOTES.txt:
```diff
@@ -0,0 +1,20 @@
+
+To get the service endpoint:
+
+If using an ingress:
```
Can you conditionally template this in/out based on whether the ingress is enabled?

viglesiasce commented on incubator/anchore-engine/templates/NOTES.txt:
```diff
@@ -0,0 +1,20 @@
+
+or to use the IP instead of service dns name:
+
+    ANCHORE_CLI_URL=$(kubectl get svc --namespace {{ .Release.Namespace }} {{ template "fullname" . }} -o jsonpath="{.spec.clusterIP}")":{{.Values.service.ports.api}}/v1/"
```
Can you add a "smoke test" here? How do users know that Anchore is installed properly? What can they do with it once its installed? Maybe also point to the user docs.

viglesiasce commented:
/hold

zhill committed:
Addresses typos and some cleanup as requested in PR review

zhill committed:
Adds more startup and config info in NOTES.txt for anchore-engine

zhill committed:
Cleanup and make labels consistent in anchore-engine deployments

zhill commented:
I have addressed the review suggestions and also added a commit to clean up some labelling issues in the deployment definitions to ensure things like 'kubectl get pods -l' filters behave as expected.

zhill committed:
Move anchore-engine chart from incubator/ to stable/

zhill commented:
Moved chart from incubator to stable for PR review per discussion with @viglesiasce

viglesiasce commented:
Worked a treat! Thanks @zhill!

/lgtm

viglesiasce commented:
/hold cancel

zhill commented:
Thanks @viglesiasce !
